custom_components/mqtt_vacuum_camera/utils/vacuum/mqtt_vacuum_services.py

```
@@ -0,0 +1,515 @@
+                service_data["payload"],
+            )
+        except Exception as e:
+            raise ServiceValidationError(f"Error sending command to vacuum: {e}")
+
+        coordinator.hass.bus.async_fire(
+            f"event_{DOMAIN}.vacuum_map_save",
+            {"topic": service_data["topic"]},
+            context=call.context,
+        )
+    except KeyError as e:
+        raise ServiceValidationError(f"Missing required parameter: {e}")
```

Ruff

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)